app/models/medium.rb

```diff
@@ -282,7 +282,16 @@ def self.lecture_search_option
   # returns search results for the media search with search_params provided
   # by the controller
   def self.search_by(search_params, page)
-    search_params[:types] = [] if search_params[:all_types] == '1'
+    # If the search is initiated from the start page, you can only get
+    # generic media sorts as results even if the 'all' radio button
+    # is seleted
+    if search_params[:all_types] == '1'
```

Comment:
we can't pass booleans as params in ruby on rails?

Reply:
A checkbox generated by the Rails form helper e.g. like `<%= f.check_box :all_types, ...%>` will by default return `'1'` for checked and `'0'` for unchecked, see [here](https://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-check_box). If you were to save the result in some `ActiveRecord` model, this would automatically translate to `true`/`false`. But here, `search_params` is just the raw hash of params that survived the whitelisting of some keys and some further manipulation going on in the media controller and is then passed to the medium model. Therefore, this value still `'1'` or `'0'`. I agree that this is realized far from optimal, as you can also see in the media controller: There is way too much business logic about the search happening there. I think that one way out would be to write a plain ruby service model that receives the params hash, deals with business logic that is right now sitting in the media controller, and in particular does things like converting '1', '0' into booleans. Such a model could then be passed to `Medium#search_by` (where again, some logic happening in that method could be made into a method in this service model). But that would possibly be something for a different PR, I think. We have previously used plain ruby models (or PORO for Plain Old Ruby) to some success in refactorization: I think `TeachableParser`, `QuizGraph` and `UserCleaner` are instances of this. There are several texts on that like[ this one](https://www.honeybadger.io/blog/refactor-ruby-rails-service-object/) or [this one](https://scottdomes.medium.com/service-objects-in-rails-75ca74214b77).

Reply:
Thanks for the links to the two blog posts. I agree that this should be something for another PR.